applications/StructuralMechanicsApplication/custom_io/gid_eigen_io.h

```diff
@@ -0,0 +1,281 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics 
+//
+//  License:		 BSD License 
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher 
+//
+
+#if !defined(KRATOS_GID_EIGEN_IO_H_INCLUDED )
+#define  KRATOS_GID_EIGEN_IO_H_INCLUDED
+
+// System includes 
+
+// External includes 
+
+// Project includes
+#include "includes/gid_io.h"
+
+
+namespace Kratos
+{
+  ///@name Kratos Globals
+  ///@{ 
+  
+  ///@} 
+  ///@name Type Definitions
+  ///@{ 
+  
+  ///@} 
+  ///@name  Enum's
+  ///@{
+      
+  ///@}
+  ///@name  Functions 
+  ///@{
+      
+  ///@}
+  ///@name Kratos Classes
+  ///@{
+  
+  /// GidIO specialized for writting Eigenvalue Results
+  
+  /** The main functionality of this class is to write the custom format 
+  * that the postprocessing of the Eigenvalues creates. 
+  * Also the result labels contain the Eigenvalue or -frequency
+  */
+class GidEigenIO : public GidIO<>
+{
+  public:
+    ///@name Type Definitions
+    ///@{
+    
+    /// Pointer definition of GidEigenIO
+    KRATOS_CLASS_POINTER_DEFINITION(GidEigenIO);
+
+    typedef std::size_t SizeType;    
+
+    typedef std::vector<std::vector<double>> EigenResults;
+    /* Structure of this EigenResults (results for one Eigenvalue)
+    Nodes
+        Coordinates
+    */
+
+    ///@}
+    ///@name Life Cycle 
+    ///@{ 
+    
+    /// Default constructor.
+    GidEigenIO( const std::string& rDatafilename,
+                GiD_PostMode Mode,
+                MultiFileFlag use_multiple_files_flag,
+                WriteDeformedMeshFlag write_deformed_flag,
+                WriteConditionsFlag write_conditions_flag) : 
+            GidIO<>(rDatafilename, 
+                    Mode, 
+                    use_multiple_files_flag, 
+                    write_deformed_flag, 
+                    write_conditions_flag) { }
+
+    /// Destructor.
+    ~GidEigenIO() = default;
+
+    // Explicitly delete the other constructors
+    GidEigenIO(const GidEigenIO&) = delete;
+    GidEigenIO& operator=(const GidEigenIO&) = delete;
+
+    ///@}
+    ///@name Operators 
+    ///@{
+    
+    
+    ///@}
+    ///@name Operations
+    ///@{
+    
+    /**
+    * Write the post-processed eigensolver-results
+    * The label is the Eigenvalue or -frequency
+    */
+    template< class TVarType>
+    inline void WriteEigenResults( ModelPart& rModelPart, 
```

Comment:
Is this not a rather large function to inline?

Reply:
+1 for this comment

Reply:
done